From 6e9fc0811f47818d9e6e9a45a6563190dd72b8c3 Mon Sep 17 00:00:00 2001
From: sicherist <sichersicher@t-online.de>
Date: Sun, 7 Jan 2018 13:59:26 +0100
Subject: [PATCH 078/249] wiki: sony: Document installation for FOTA-only
 devices

 * The current instructions on the LineageOS Wiki do not work
   for pre-shinano Sony devices. These devices have no separate
   recovery partition and need a different method to install TWRP

 * Optionally document the TWRP recovery to boot installation,
    followed by the FOTA permament recovery installation

 * Use the install_variant new "sony_init_fota" key for
    all the documented and concerned Sony devices

Change-Id: If9d6fc27e28a8005f2fa4a27f6ec5cf5c5f26a53
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 _data/devices/dogo.yml                        |  1 +
 _data/devices/hayabusa.yml                    |  1 +
 _data/devices/huashan.yml                     |  1 +
 _data/devices/mint.yml                        |  1 +
 _data/devices/nicki.yml                       |  1 +
 _data/devices/odin.yml                        |  1 +
 _data/devices/pollux.yml                      |  2 +-
 _data/devices/pollux_windy.yml                |  2 +-
 _data/devices/taoshan.yml                     |  1 +
 _data/devices/tsubasa.yml                     |  1 +
 _data/devices/yuga.yml                        |  1 +
 .../recovery_install_fastboot_sony.md         | 55 +++++++++++++++++++
 12 files changed, 66 insertions(+), 2 deletions(-)

diff --git a/_data/devices/dogo.yml b/_data/devices/dogo.yml
index f745423..f82589b 100644
--- a/_data/devices/dogo.yml
+++ b/_data/devices/dogo.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 131.3 mm (5.17 in)
 image: dogo.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_apq8064
 maintainers: [Daedroza, Chippa_a, CyberWalkMaN, Adrian DC]
 name: Xperia ZR
diff --git a/_data/devices/hayabusa.yml b/_data/devices/hayabusa.yml
index b86b8f1..878d20b 100644
--- a/_data/devices/hayabusa.yml
+++ b/_data/devices/hayabusa.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 131.0 mm
 image: hayabusa.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8x60
 maintainers: [Adrian DC]
 name: Xperia TX
diff --git a/_data/devices/huashan.yml b/_data/devices/huashan.yml
index 7125059..4b46215 100644
--- a/_data/devices/huashan.yml
+++ b/_data/devices/huashan.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 133 mm
 image: huashan.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8960t
 maintainers: [Adrian DC]
 name: Xperia SP
diff --git a/_data/devices/mint.yml b/_data/devices/mint.yml
index d17691a..abeacc3 100644
--- a/_data/devices/mint.yml
+++ b/_data/devices/mint.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 129.4 mm
 image: mint.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8x60
 maintainers: [Adrian DC]
 name: Xperia T
diff --git a/_data/devices/nicki.yml b/_data/devices/nicki.yml
index 1062040..dd4c02b 100644
--- a/_data/devices/nicki.yml
+++ b/_data/devices/nicki.yml
@@ -16,6 +16,7 @@ gpu: Adreno 305
 has_recovery_partition: true
 image: nicki.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8x27
 maintainers: []
 name: Xperia M
diff --git a/_data/devices/odin.yml b/_data/devices/odin.yml
index 9a1e0e5..1137569 100644
--- a/_data/devices/odin.yml
+++ b/_data/devices/odin.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 131.7 mm (5.18 in)
 image: odin.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_apq8064
 maintainers: [CyberWalkMaN, Daedroza, Chippa_a, Adrian DC]
 name: Xperia ZL
diff --git a/_data/devices/pollux.yml b/_data/devices/pollux.yml
index ce2f3cb..844a7e2 100644
--- a/_data/devices/pollux.yml
+++ b/_data/devices/pollux.yml
@@ -18,7 +18,7 @@ has_recovery_partition: true
 height: 172 mm (6.77 in)
 image: pollux.png
 install_method: fastboot_sony
-install_variant: sony_unlock_contacts
+install_variant: sony_unlock_contacts sony_init_fota
 kernel: android_kernel_sony_apq8064
 maintainers: [CaHbKaUp, CyberWalkMaN, Daedroza, Chippa_a, Adrian DC]
 name: Xperia Tablet Z LTE
diff --git a/_data/devices/pollux_windy.yml b/_data/devices/pollux_windy.yml
index 5e4705c..74698df 100644
--- a/_data/devices/pollux_windy.yml
+++ b/_data/devices/pollux_windy.yml
@@ -18,7 +18,7 @@ has_recovery_partition: true
 height: 172 mm (6.77 in)
 image: pollux.png
 install_method: fastboot_sony
-install_variant: sony_unlock_contacts
+install_variant: sony_unlock_contacts sony_init_fota
 kernel: android_kernel_sony_apq8064
 maintainers: [CaHbKaUp, CyberWalkMaN, Daedroza, Chippa_a, Adrian DC]
 name: Xperia Tablet Z Wi-Fi
diff --git a/_data/devices/taoshan.yml b/_data/devices/taoshan.yml
index 466e9a5..c499d8b 100644
--- a/_data/devices/taoshan.yml
+++ b/_data/devices/taoshan.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 128.7 mm (5.07 in)
 image: taoshan.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8930
 maintainers: [corphish]
 name: Xperia L
diff --git a/_data/devices/tsubasa.yml b/_data/devices/tsubasa.yml
index 914d35c..dc55a12 100644
--- a/_data/devices/tsubasa.yml
+++ b/_data/devices/tsubasa.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 129.0 mm
 image: tsubasa.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_msm8x60
 maintainers: [Adrian DC]
 name: Xperia V
diff --git a/_data/devices/yuga.yml b/_data/devices/yuga.yml
index fba5242..2a57d15 100644
--- a/_data/devices/yuga.yml
+++ b/_data/devices/yuga.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 139 mm (5.47 in)
 image: yuga.png
 install_method: fastboot_sony
+install_variant: sony_init_fota
 kernel: android_kernel_sony_apq8064
 maintainers: [Daedroza, CyberWalkMaN, MardonHH, Chippa_a, Adrian DC]
 name: Xperia Z
diff --git a/_includes/templates/recovery_install_fastboot_sony.md b/_includes/templates/recovery_install_fastboot_sony.md
index 05078dd..0dd94c2 100644
--- a/_includes/templates/recovery_install_fastboot_sony.md
+++ b/_includes/templates/recovery_install_fastboot_sony.md
@@ -34,4 +34,59 @@ If you wish to backup the TA partition first, you can find tutorials related to
 6. Follow the instructions on [Sony's official unlocking website](http://developer.sonymobile.com/unlockbootloader/unlock-yourboot-loader/) to unlock your bootloader.
 7. Since the device resets completely, you will need to re-enable USB debugging to continue.
 
+{% if device.install_variant and device.install_variant contains "sony_init_fota" %}
+
+{% if site.data.devices[page.device].custom_twrp_codename %}
+{% assign twrp_codename = site.data.devices[page.device].custom_twrp_codename %}
+{% else %}
+{% assign twrp_codename = site.data.devices[page.device].codename %}
+{% endif %}
+
+## Installing a custom recovery using `fastboot`
+
+{% if site.data.devices[page.device].custom_twrp_link %}
+1. Download a custom recovery - you can download [TWRP]({{ site.data.devices[page.device].custom_twrp_link }}).
+{% else %}
+1. Download a custom recovery - you can download [TWRP](https://dl.twrp.me/{{ twrp_codename }}). Simply download the latest recovery file, named something like `twrp-x.x.x-x-{{ twrp_codename }}.img`.
+{% endif %}
+2. Connect your device to your PC via USB.
+3. On the computer, open a command prompt (on Windows) or terminal (on Linux or macOS) window, and type:
+
+        adb reboot bootloader
+
+    {% if site.data.devices[page.device].download_boot %}
+    You can also boot into fastboot mode via a key combination:
+
+    * {{ site.data.devices[page.device].download_boot }}
+    {% endif %}
+4. Once the device is in fastboot mode, verify your PC finds it by typing:
+
+        fastboot devices
+
+    {% include tip.html content="If you see `no permissions fastboot` while on Linux or macOS, try running `fastboot` as root." %}
+5. Temporarily flash TWRP to `boot`:
+
+        fastboot flash boot twrp-x.x.x-x-{{ twrp_codename }}.img
+
+    {% include tip.html content="The file may not be named identically to what stands in this command, so adjust accordingly. Remember to adjust the filename in the following commands as well." %}
+6. Reboot to the TWRP recovery:
+
+        fastboot reboot
+7. Push the TWRP image to your device:
+
+        adb push twrp-x.x.x-x-{{ twrp_codename }}.img /sdcard
+8. Enter shell on the device:
+
+        adb shell
+9. Flash TWRP to `recovery` permanently:
+
+        dd if=/sdcard/twrp-x.x.x-x-{{ twrp_codename }}.img of=/dev/block/platform/msm_sdcc.1/by-name/FOTAKernel
+10. Exit the adb shell:
+
+        exit
+
+{% else %}
+
 {% include templates/recovery_install_fastboot_generic.md %}
+
+{% endif %}
-- 
2.17.1

